### ISSUE TITLE
Create a CLI (`mariactl`) to abstract backup and restore commands 

### ISSUE THREAD

**Issue by mmontes11** [mariadb-operator/mariadb-operator#16]
Instead of creating commands by manipulating strings in the `backupcmd` pkg, create a CLI to better abstract them without needing to rely on bash at all.

Requirements:
- [spf13/cobra](https://github.com/spf13/cobra)
- Multi-arch image based on alpine

**Comment by ibotty** [comment]
I usually prefer [kubeconfig plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) because they get namespaced.  Would you consider this?

**Comment by mmontes11** [comment]
Hey @Trojan295 ! Thanks a lot for your interest in contributing to `mariadb-operator` 🦭 ❤️ 

Unfortunately, we recently dropped the `mariabackup` support for now, as we were having a lot of issues with our newest HA support. We are planning to support it back, but a few refactors need to be done first, see this for more detail:
- https://github.com/mariadb-operator/mariadb-operator/pull/97

Regarding the CLI, we have decided to de-prioritize this item (see the new [ROADMAP.md](https://github.com/mariadb-operator/mariadb-operator/blob/main/ROADMAP.md)) since there isn't too much to abstract now and the CRDs APIs might change during the refactor I've mentioned. In any case, I still see the value of a CLI from the user experience perspective, it's something that other operators do and that cloud native end users would expected.

I would suggest you stay tuned, as this or other features will require new contributiors at some point. 🚀
